src/Umbraco.Web.UI.Client/src/packages/core/collection/default/collection-default.element.ts

```diff
@@ -69,6 +69,15 @@ export class UmbCollectionDefaultElement extends UmbLitElement {
 		);
 	}
 
+	#generateEmptyStateLabel() {
+		const labelKey =
+			this.#collectionContext?.manifest?.meta.noItemsLabel ??
+			this.#collectionContext?.getConfig()?.noItemsLabel ??
+			'#collection_noItemsTitle';
```

Comment:
We should move this responsibility to one method on the context. Just for good separation of concern, avoiding logic of data located elsewhere. but put the logic where the data belongs :-)